### PR TITLE
AR-1392: Handling mixed-case search terms + tests to verify

### PR DESCRIPTION
**Pull request by timharrison-moj**
This PR 

* Fixes a couple of issues with keyword and prisoner-detail searches when mixed-case search terms are presented & introduces some test cases to verify these are now handled ok.  
* Updates the Swagger descriptions for keyword and prisoner detail searches.
* Fixes a bug where search terms were not correctly split for keyword searches.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerDetailService.kt [hunk 1]
```diff
@@ -91,21 +91,21 @@ class PrisonerDetailService(
       firstName.takeIf { !it.isNullOrBlank() }?.let {
         detailQuery.must(
           QueryBuilders.boolQuery()
-            .should(QueryBuilders.matchQuery("firstName", it).fuzzyTranspositions(fuzzyMatch).boost(5f))
-            .should(QueryBuilders.matchQuery("aliases.firstName", it).fuzzyTranspositions(fuzzyMatch).boost(2f))
-            .should(QueryBuilders.wildcardQuery("firstName", it).boost(5f))
-            .should(QueryBuilders.wildcardQuery("aliases.firstName", it).boost(2f))
+            .should(QueryBuilders.matchQuery("firstName", it.lowercase()).fuzzyTranspositions(fuzzyMatch).boost(5f))
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerDetailService.kt:94]: I'm curious, why did you need to lowercase? These aren't keywords

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerDetailService.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerDetailService.kt:94]: Thanks Andy. It was a problem encountered whilst testing in dev -  names with mixed case in the search terms e.g. `Ninice` found nothing but when in lowercase e.g. `ninice` it finds the matches. The index actually has them stored as uppercase. A quick conversion of all the terms to lowercase except keyword types (which are uppercased) resolved the issue. Reading a bit further into it it looks like it is expecting all non-keyword terms to be provided in lowercase, and then it will match them with *any* case stored in the index. Agree its a bit odd though.
